Code/src/main/java/nl/utwente/group10/ui/components/anchors/ConnectionAnchor.java

```diff
@@ -79,17 +95,15 @@ public boolean isConnected() {
      */
     public abstract void disconnectFrom(Connection connection);
 
-    public Optional<ConnectionAnchor> getOtherAnchor() {
+    public Optional<? extends ConnectionAnchor> getOtherAnchor() {
```

Comment:
Wat je zou kunnen doen is `getOtherAnchor` in de subklassen overschrijven, bijv. in OutputAnchor:

    @Override
    public Optional<InputAnchor> getOtherAnchor() {
        return (Optional<InputAnchor>) super.getOtherAnchor();
    }

...maar je kunt ook zeggen dat dat een lelijke unchecked cast is en bovendien nergens voor nodig. Your call.